src/Common/src/Interop/Unix/libcurl/Interop.SafeCurlHandle.cs

```diff
@@ -37,6 +40,19 @@ protected override bool ReleaseHandle()
 
         internal sealed class SafeCurlMultiHandle : SafeHandle
         {
+            private readonly CancellationTokenSource _cancellationTokenSource = new CancellationTokenSource();
+            private readonly List<int> _fdList = new List<int>();
```

Comment:
Have replaced with HashSet and lazy allocation. @davidsh , @SidharthNabar , any idea about typical no. of outstanding requests per HttpClientHandler?